docs/sphinx/reference-libobs-util-darray.rst

```diff
@@ -50,6 +50,42 @@ dynamic array value with a reference (&) operator.  For example:
    /* free when complete */
    da_free(array_of_integers);
 
+To pass dynamic arrays to functions as parameters, create a typedef with the
+:c:macro:`DARRAY` macro and type so you can pass the dynamic array with a
+reference (&) operator. An alternative is to define a struct that will contain
+the dynamic array, and that struct will be passed to the functions, instead of
+the dynamic array directly. An example with the typedef method:
+
+.. code:: cpp
+
+   typedef DARRAY(int) int_array_t;
+
+   void generate_integers(int_array_t *integers, int start, int end)
+   {
+           for (int i = start; i < end; i++)
+                   da_push_back(*integers, &i);
```

Comment:
For this, I removed the parentheses that is necessary atm, like `da_push_back((*integers), &i);`, due to conversation at https://github.com/obsproject/obs-studio/pull/9072#discussion_r1236385202